docs/_components/dropzone.md

```diff
@@ -0,0 +1,638 @@
+
+{% raw %}
+```twig
+{% import '@pulsar/pulsar/v2/helpers/html.html.twig' as html %}
+
+{{
+    html.dropzone({
+        'id': 'drop-zone'
+    })
+}}
+```
+```javascript
+import DropZoneComponentFactory from '../path/to/DropZoneComponentFactory';
+
+const dropZoneComponent = DropZoneComponentFactory.create(
+    // <html> element
+    document.documentElement, 
+    // selector to identify DropZone elements
+    '.dropzone'
+);
+
+dropZoneComponent.init();
+```
+{% endraw %}
+
+---
+
+### Passive
+
+DropZone initiated in passive mode.
+
+![DropZoneComponent passive drop]({{ site.baseurl }}/assets/image_examples/dropzone-passive-drop.gif){: .dropzone-example }
+
+{% raw %}
+```html
+<div class="dropzone" data-dropzone-passive="true">
+    <p>Node to be converted to a DropZone</p>
+</div>
+```
+```javascript
+import DropZoneComponentFactory from '../path/to/DropZoneComponentFactory';
+
+const dropZoneComponent = DropZoneComponentFactory.create(
+    // <html> element
+    document.documentElement, 
+    // selector to identify DropZone elements
+    '.dropzone'
+);
+
+dropZoneComponent.init();
+```
+{% endraw %}
+
+---
+
+### Enriched passive 
+
+An example of a passive DropZone leveraging the DropZoneComponent's callbacks and public methods (see below) to trigger help state changes and a backdrop.
+
+![DropZoneComponent passive enriched]({{ site.baseurl }}/assets/image_examples/dropzone-passive-enriched.gif){: .dropzone-example }
+
+---
+
+### Submission
+
+In order to use a DropZone as a file input for a form, it is required that the form is submitted as a XHR / Ajax request. This is necessary as we need to be able to query the DropZoneComponent API to retrieve files before sending the form.
+
+```javascript
+const data = new FormData(form);
+const id = parseInt(dropZoneHtmlNode.getAttribute('data-dropzone-id'));
+const file = dropZoneComponent.getFilesFromDropZone(id, 0);
+
+data.append('dropzone-file', file.raw);
+
+// ...
+```
+
+When retrieving files from a DropZone instance, you should expect the following schema.
+
+```
+{
+    id: [Number], // a unique identifier
+    raw: [FileObject], // a reference to the original FileObject
+    name: [String], // file name
+    type: [String], // file type
+    size: [String], // file size
+    thumbnail: [String|false], // file thumbnail
+    meta: [Object] // additional file information
+}
+```
+
+---
+
+## Options
+
+Options can be initialised as part of the `init({...options})` method, or as attributes on DropZone HTML elements. Below are a list of configurable options with an environment table showing initiation examples.
+
+Any option from the reference above that is prefixed with `data-dropzone-` can be initialised using the twig helper (or HTML when in passive mode) or through the DropZone's `init(...)` method as a camel case version of the attribute.
+
+---
+
+#### Max Files
+
+```
+@param {String|Number}
+@deafult [5]
```

Comment:
I've made up my own syntax here, this is intended to be the default value of that option, maybe I should drop the pseudo JSDoc here all together if it's confusing.